src/test/fuzz/process_message_e2e.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2020-2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <consensus/consensus.h>
+#include <net.h>
+#include <net_processing.h>
+#include <primitives/transaction.h>
+#include <protocol.h>
+#include <script/script.h>
+#include <serialize.h>
+#include <span.h>
+#include <streams.h>
+#include <sync.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/fuzz/util/net.h>
+#include <test/util/mining.h>
+#include <test/util/net.h>
+#include <test/util/setup_common.h>
+#include <test/util/validation.h>
+#include <util/chaintype.h>
+#include <util/check.h>
+#include <util/time.h>
+#include <validation.h>
+#include <validationinterface.h>
+#include <version.h>
+
+
+#include <atomic>
+#include <cstdlib>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <string_view>
+#include <vector>
+
+// There are two sets of tests:
+// - direct: pass the fuzzed message directly to PeerManager::ProcessMessage() (fuzz/process_message.cpp)
+// - e2e: send the fuzzed message to a mocked socket which is read from Connman
+//   and processed by PeerManager (like it is happening in the "real world") (fuzz/process_message_e2e.cpp)
+
+namespace {
+NetTestingSetup* g_setup;
+std::string_view LIMIT_TO_MESSAGE_TYPE{};
+} // namespace
+
+void initialize_process_message_e2e()
+{
+    if (const auto val{std::getenv("LIMIT_TO_MESSAGE_TYPE")}) {
+        LIMIT_TO_MESSAGE_TYPE = val;
+        Assert(std::count(getAllNetMessageTypes().begin(), getAllNetMessageTypes().end(), LIMIT_TO_MESSAGE_TYPE)); // Unknown message type passed
+    }
+
+    static const auto testing_setup = MakeNoLogFileContext<NetTestingSetup>(
+            /*chain_type=*/ChainType::REGTEST,
+            /*extra_args=*/{"-txreconciliation"});
+    g_setup = testing_setup.get();
+    for (int i = 0; i < 2 * COINBASE_MATURITY; i++) {
```

Comment:
Done.

I changed it to use `{}` as suggested but I don't see a point to use `{}` instead of `=` for initialization when `auto` is used. I find the `{}` variant a little bit less readable but see the merit to use it when `Type1 x{expresson_of_Type2};` is used to detect incompatibilities between `Type1` and `Type2`. But why do that for `auto x{expressoin_of_any_type};`?

Reply:
I am still not convinced about `auto` and brace-initialization.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es23-prefer-the--initializer-syntax mentions:
> Use = only when you are sure that there can be no narrowing conversions
> For built-in arithmetic types, use = only with auto

There will not be narrowing conversions with `auto x = expression;`

One of the examples down there uses `auto p = ...`.